### DISCUSSION TITLE
SSH Issues

### DISCUSSION THREAD

**Discussion by veetoah** [gtxaspec/wz_mini_hacks#502] [Q&A] [answered]
Hi,

Following guide for v3 and have watched video on ssh key authentication but I continue to receive connection refused. Can someone let me know which of my steps are wrong ?

Copied sd root folder to sd card , so as soon as you click on drive it has wz_mini folder and factory file.

Created SSH key (did not notice any other relevant steps on guide to follow)

Copied entire public SSH key I created (including part that says ssh rsa at the beginning and one that includes pc name at end) underneath where it says place authorized keys below

used putty with ip address and default ssh port 22

used terminal with computer ssh (pc username - one listed under home folder)@ip address, ssh root@ipaddress and I never get the enter passphrase note.

**Comment by Bob-AOL** [comment]
Do you actually receive a response?

**Reply by Bob-AOL** [reply]
How and where did you create your public key and how and where did you put it on the camera?

Putty can get a little busier than necessary sometimes. And remember you're logging into the camera as 'root'. Try just using a Windows admin command prompt for the session --  ssh root@wyzecamip and and see what you get.

**Reply by Bob-AOL** [reply]
After reading your post again I kinda sorted out the answers to my questions. So did you stipulate a passphrase when creating the key pair?
And again, what response are you actually receiving from the camera?

**Reply by veetoah** [reply]
Yes, I created a passphrase. I tried windows with same result as soon as I enter the above ssh root@wyzecam I get a connection refused error. The camera is functional and showing up on my wyze app.

**Comment by bjs-pdx** [comment]
Are you using Windows?  Did it default to adding CR/LF characters in your key?  Make sure all the key data is only one line, not multiple in the key file on the camera.

**Reply by veetoah** [reply]
I tried windows with the same result. Not sure what you mean by CR/LF characters but I just followed the guide and created a key with a passphrase via ssh-keygen. I checked the key file on the camera and all is in one line (it was directly copied from key generated file on my laptop and tried to connect with my laptop and a seperate pc

**Comment by bjs-pdx** [comment]
CR=carriage return  LF=line feed.  If you look at the key file that you copied to the SD card and it has multiple lines, then you probably have CR/LF characters in the file.  It wraps a long text line into multiple text lines.  Linux doesn't do that.  If it is multiple lines, then it won't work.  You should have one, long line in the file with all the information for the key.  For example:

xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
vs
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
 
How did you copy the key into the file on the sd card?

**Reply by veetoah** [reply]
I went into my id.rsa pub file and copied the text and pasted onto the keyfile in wyze under where it says paste key below. The text appears to all be one line as I drag the text editor to fit my screen it fills up most of it and then drops to another line. I put the sd card in windows and it will only show me the .img file and not the root folder. Are you saying I should re-format using windows instead ?

**Reply by bjs-pdx** [reply]
Not sure what you mean by .img file.  There should be just these in the root of the sd card:

![image](https://user-images.githubusercontent.com/14082937/228680176-1c9ed6e0-3de7-4d3a-a9dc-b738deb00c45.png)

**Reply by veetoah** [reply]
I don't have a demo.bin file. On linux I show all except demo.bin but when I pop in the sd card into windows I only show the factory file.

**Reply by bjs-pdx** [reply]
Again, not sure what you mean by factory file.  Can you post a screenshot?

**Reply by veetoah** [reply]
referring to your screenshot the file name that starts with factory. I don't have a demo.bin file and its not showing within the zip I downloaded either.  let me know if this helps

**Reply by bjs-pdx** [reply]
Have you tried starting from scratch?  Sounds to me like you are missing a step somewhere.

**Reply by Bob-AOL** [reply]
Start over and keep it simple.

Generate a new key pair from a Windows command prompt using ssh-keygen. Save it in your .ssh/id_rsa file. Don't mess with the passphrase. Windows will not add any formatting to the text. It is a plain text file that will work just fine.

![AuthKey Unix(LF)](https://user-images.githubusercontent.com/110629141/228680446-128f98fb-aede-45e6-94ea-8231310fefe6.jpg)

Copy the key to the etc/ssh authorized keys file on the SD card. You're done with the keys.

Now check the .ssh known hosts file in your laptop for orphan entries and clean them out. 

Try a new session with the camera. It should respond with a request to be added as a known host. You'll obviously want to accept. If successful, you'll then be logged into the camera as root.

**Reply by Bob-AOL** [reply]
Demo.bin is a firmware flash file. Only wz_mini and the factory camera I.D. file should initially be on the SD card. It will become populated with other files during normal operation, but the two files mentioned are the only ones that should be on the SD card for the initial bootup of mini_hacks.

**Reply by veetoah** [reply]
Just tried the above on a fresh windows pc and I'm still receiving a connection refused error. (opened command prompt as admin and ssh root@ip address). I'm at a loss as I'm able to ssh into other items. edit: I even tried using a different  random usernames @ip and they all receive the same error

**Reply by Bob-AOL** [reply]
Did you check your known hosts file? I should have been more specific about that and wasn't considering _good_ known hosts when suggesting to clean the entries.

Are (were?) there any entries for the address of the camera you are trying to reach?

**Reply by veetoah** [reply]
I don't have a known hosts file.. checked within c/users/username/ssh . only have two files id_rsa and id_rsa.pub

**Reply by Bob-AOL** [reply]
Are you sure the camera is even booting from the SD card?

**Reply by veetoah** [reply]
how would I determine this ? The camera is still fully functional after it turns on with the sd card inside

**Reply by Bob-AOL** [reply]
Did you activate/enable any wz_mini features for testing?

**Reply by veetoah** [reply]
everything is stock. wyze cam using latest firmware and wz mini with no modifications

**Reply by Bob-AOL** [reply, marked as answer]
Ok. Pull the SD card and pop it into something for editing the wz_mini.conf file to at least enable the web server on the camera..., (assuming the camera is even seeing the SD card).Then see if the camera web server is accessible by simply entering the IP address into a browser. Or..., just do a port scan on the camera IP address to see if port 22 is open.

There should also be additional files now present on the SD card  for examination, but that's a whole other bag o' snakes better left for another discussion.

**Reply by Bob-AOL** [reply]
Uhhh...., what latest firmware? The _latest_ Wyze firmware will not work and has even been called back by Wyze in favor of the last known good version.. Anything beyond 4.36.9.139 is not compatible with Mini-Hacks.

**Reply by veetoah** [reply]
ok so good news is when it upgraded firmware it went to 4.36.9.139 
I set the camera web server to true and the ip address of the cam is not found. Its almost as if the camera is not even booting from the sd card...

**Reply by Bob-AOL** [reply]
That's why I asked. Does the Wyze app show the SD card as being present in the camera?

**Reply by veetoah** [reply]
misunderstood was unaware you could check app. app is saying there is no microsd card available

**Reply by Bob-AOL** [reply]
Oops. Can't help ya there, bro.

Basic 32GB SDHC Class 10 FAT32 micro SD card is all wz_mini needs and what the camera should be happy with to get Mini_Hacks off the ground for you..., provided the card reader is working and the card is inserted correctly.

Good luck.

**Reply by veetoah** [reply]
well looks like you helped diagnose the problem. I tried a different sd card and was able to finally get root!! Now next problem lol.. I am trying to connect via rtsp and set test as the password to attempt to login. I typed the following via vlc network stream but receiving error vlc unable to find mrl.  rtsp://admin:test@my IP_ADDRESS:8554/unicast

**Reply by Bob-AOL** [reply]
Looks like it. You're welcome.

The username and password need to be the camera's wifi login credentials.

**Reply by veetoah** [reply]
so the actual wifi ssid and password right ?  I entered both of those in the config file and also in the rtsp address and still receiving same error.

**Reply by Bob-AOL** [reply]
Actually, no. Was thinking of something else completely unrelated. My bad.

The user/pass combination can be arbitrarily set in the config. It simply needs to match in the MRL (Media Resource Locator) address used to access the stream. (The login credential requirement can also be disabled in the config if desired.)

You obviously also need to have RTSP enabled and properly configured for the username and password to have something to authenticate.

**Reply by veetoah** [reply]
looks like oversight on my part need to also set hi or lo stream to true. got it up and running, thanks for all the support here!!!

**Reply by Bob-AOL** [reply]
Good deal. You should tag the thread as answered for anyone else possibly looking for a solution to the same problem (set of problems?)
